src-ui.v3/src/Main.hs

```diff
@@ -396,12 +402,30 @@ bodyElement4 = mdo
             pure $ tagPromptlyDyn tVal addResult
           pure ()
         
-        let evReports' = updated (_dropdown_value ddReports)
-            dynIdxSt   = ddReports ^. dropdown_value
-
+        let evReports'   = updated (_dropdown_value ddReports)
+            dynIdxSt     = ddReports ^. dropdown_value
+            evIdxChange  = updated dynIdxSt --ddReports ^. dropdown_change
+        _ <- mdo
+          historyState <- manageHistory $ HistoryCommand_PushState <$> setState
```

Comment:
I think that calling manageHistory is better suited to https://github.com/haskell-CI/hackage-matrix-builder/pull/65/files#diff-46676349586994bc32e17ce7fa5eb9e0R107 . That way you encompass the setting and reading of everything in the one place. 

To set the route from anywhere at that point, you'll want an EventWriter that you can push an event onto to change the event that we can pull out all the way up the top and feed it into manageHistory. To do this nicely, you'll want to newtype the EventWriter like in obelisk. Put this in a separate file because you'll need a boatload of instances for that newtype to make it work in here. See https://github.com/obsidiansystems/obelisk/blob/develop/lib/route/src/Obelisk/Route/Frontend.hs#L271 You'll want the SetRouteT and the SetRoute Typeclass so that you can make a nice syntax for setting the route in your widgets. 

Putting the manageHistory all the way up the top centralises the routing in a pure FRP way rather than using the dom as a mutable variable and having the SetRoute eventwriter through all of your widgets means it is now easy to change the route from the search box. :smile: